public/app/features/query/components/QueryEditorRows.tsx

```diff
@@ -90,7 +85,9 @@ export class QueryEditorRows extends PureComponent<Props> {
           hide: item.hide,
           datasource: dataSourceRef,
         };
-      })
+      }),
+      dataSource,
+      index
```

Comment:
not sure about this, this feels like a halfway implementation (where the consumer of the implementation needs to handle some logic, while some is inside the component. 

I think we either make QueryEditorRows have callbacks for onQueryChange (query, index) (so basically forward the callback from QueryEditorRow.  and an onChangeDataSource which we also just forward (query, dsSettings, index). And the consumer of the component needs to handle both (QueryGroup and QueryRows). Or we keep the current interface with just onChangeQuery and the consumer needs to figure out if there was a datasource change or not.